### PR TITLE
Add guarded batch functions in Gateway contract

### PR DESCRIPTION
**Pull request by marie-fourier**
## Description

- created the GatewayV2 contract with guarded batch functions
`function sendBatchGuarded`
`function sendBatchFromAccountGuarded`
`function delegateBatchGuarded`
`function delegateBatchWithGasPriceGuarded`
`function delegateBatchesGuarded`

- Wrote tests for GatewayV2 `(100% Statements 43/43, 80.95% Branches 34/42, 100% Functions 19/19, 100% Lines 49/49)`
- Removed `.only` from HopFacet's tests
- Deployed GatewayV2 on testnets `(Goerli, Mumbai, BscTest, FantomTest, ArbitrumNitro, Fuji and Sokol)`
- Deployed BalancesHelperV2 on Fuji and Sokol

## Motivation and Context

-

## How Has This Been Tested?

- unit tests
-
-

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/gateway/Gateway.sol [hunk 1]
```diff
@@ -17,7 +18,7 @@ import "../personal/PersonalAccountRegistry.sol";
  *
  * @author Stanisław Głogowski <stan@pillarproject.io>
  */
-contract Gateway is Initializable, SignatureValidator {
+contract Gateway is Initializable, SignatureValidator, Guarded {
```
Comment (zincoshine) [inline src/gateway/Gateway.sol:21]: Create a new version of Gateway called GatewayV2

**Review by zincoshine — CHANGES_REQUESTED** [review]
It is better to implement this change in a new contract, which will make it easier to switch between the two if and when needed.

File: src/gateway/Gateway.sol (hunk 1 above)
Comment (marie-fourier) [reply src/gateway/Gateway.sol:21]: agree. will do this

**Comment by marie-fourier** [comment]
@zincoshine HopFacet.ts changed because I removed a typo from there (updated in the pr description)

**Comment by zincoshine** [comment]
Pls deploy on gnosis testnet, avax testnets as well

**Review by zincoshine — APPROVED** [review]

**Review by lbw33 — APPROVED** [review]

**Pull request merged**